src/algebra/group/units.lean

```diff
@@ -400,6 +404,29 @@ begin
   simp [h.unit_spec]
 end
 
+/-- An element `a : M` of a monoid has a left inverse
+if there is some `x : M` satisfying `x * a = 1`. -/
+def has_left_inv [monoid M] (a : M) : Prop := ∃ x : M, x * a = 1
+
+/-- An element `a : M` of a monoid has a right inverse
+if there is some `x : M` satisfying `a * x = 1`. -/
+def has_right_inv [monoid M] (a : M) : Prop := ∃ x : M, a * x = 1
+
+/-- An element of a monoid is a unit
+if and only if it has both a left inverse and a right inverse. -/
+theorem is_unit_iff_has_left_inv_right_inv [monoid M] {a : M} :
+  is_unit a ↔ has_left_inv a ∧ has_right_inv a :=
+⟨λ h, ⟨h.exists_left_inv, h.exists_right_inv⟩,
+  λ ⟨⟨x, hxa⟩, ⟨y, hay⟩⟩, ⟨⟨a, x, by { convert hay,
+  rw [←mul_one x, ←hay, ←mul_assoc, hxa, one_mul] }, hxa⟩, rfl⟩⟩
+
+/-- An element of a monoid is a unit
+if it has a left inverse which also has a left inverse. -/
+theorem is_unit_of_has_left_inv_of_has_left_inv [monoid M] {a b : M}
```

Comment:
```suggestion
theorem is_unit_of_left_inv_of_has_left_inv [monoid M] {a b : M}
```
since the first hypothesis isn't in terms of `has_left_inv`

Reply:
Thanks! I changed the name.